ENH: InterestAreas object

teonbrooks opened the pull request:
This object reorganizes the data into a trial by interest area format to look at properties across trials with ease.
Test needed.

teonbrooks committed:
WIP

teonbrooks committed:
WIP

teonbrooks committed:
ENH: Add functioning InterestAreas

Eric89GXL commented:
Please also add an example, since I have no idea what an interest area is :)

Eric89GXL commented:
You'll need to get `OrderedDict` from mne-python

dengemann commented on pyeparse/ia.py:
```diff
@@ -0,0 +1,157 @@
+                            % type(raw))
+
+        trial_ids = raw.find_events('TRIALID', 1)
+        self.n_trials = n_trials = trial_ids.shape[0]
```
should we stick with 'eopchs' for consistency with MNE?

teonbrooks replied:
like `n_epochs`?

larsoner commented on pyeparse/ia.py:
```diff
@@ -0,0 +1,157 @@
+
+
+def read_ia(filename):
+    ia = open(filename).readlines()
```
booo hissss
`with open`

teonbrooks replied:
lol, what's the diff?

larsoner replied:
`with open():` will ensure the file gets closed when you are done with it. On Py3k you get `ResourceWarning`s in some cases if you don't do it.

teonbrooks replied:
ah, got it.

dengemann commented:
>Please also add an example, since I have no idea what an interest area is :)

+1 I have no idea what this is good for :) I'm curious! Sounds like an interesting area of development ... ;)

teonlamont commented:
sure, sure :) examples are on the way. InterestAreas are at the center of lots of reading research and cognitive tasks in general. They are essentially the ROIs of reading. Lots of reading measures such as gaze duration, single fixation duration, first-fixation duration, first-pass reading, etc. are defined wrt a fixation or fixations, their order, and location. Gaze duration is defined as the sum of fixations from the first time you enter a region until you exit in any direction whereas go-past is defined as the sum of fixations from the first time you enter the region until you make a right-bound exit to the following word. 

There are lots of research that model these different measures to different cognitive proccesses but the backbone of them all are defined in Interest Areas.

For reading, they are usually rectangular bounding boxes around the words in the sentence. They are parsed with SR-Research Experiment Builder. I used their template as a basis.

teonlamont commented:
real example with code to come.

example for my experiment I'm working on now:

![trial](https://cloud.githubusercontent.com/assets/1578674/7672345/b5cd8a6a-fcef-11e4-9187-c87bd8f6b80e.jpg)

teonlamont commented:
@Eric89GXL I tried `git grep OrderedDict` for mne-python I couldn't find any mentions

dengemann commented:
from collections import OrderedDict

> On 18 May 2015, at 12:54, Teon L Brooks <notifications@github.com> wrote:
> 
> @Eric89GXL I tried git grep OrderedDict for mne-python I couldn't find any mentions
> 
> —
> Reply to this email directly or view it on GitHub.
>

teonlamont commented:
I already have that here: https://github.com/pyeparse/pyeparse/pull/49/files#diff-7489481660ad1bad504b865f7f0982bcR8 but the build crashed

Eric89GXL commented:
we must not support 2.6 on mne-python, take it from `vispy` then, it uses a standard Python 2.6 replacement

teonbrooks committed:
FIX example

teonbrooks committed:
ENH optimize ia

There’s now a dependency on pandas with this module.

Add pandas dependency error if someone doesn’t have Pandas when
executing functions

teonbrooks committed:
ENH: ia, FIX ordereddict import

OrderedDict support for py < 2.7

teonbrooks committed:
WIP

This is needed for current analyses, fix and squash later

teonbrooks committed:
ENH explicit name

teonbrooks committed:
WIP ReadingMixin

teonbrooks committed:
FIX

teonbrooks committed:
wip

Restructuring InterestAreas

teonbrooks committed:
more wip

teonbrooks committed:
wip

teonbrooks committed:
Interest Areas overhaul

teonbrooks committed:
ADD Reading class

teonbrooks committed:
ENH cleanup, add notes

teonbrooks committed:
Rename to interest_area, Add reading module

Make a general InterestArea class
Make a module dedicated for reading measures.
WIP on the docstring and methods.

teonbrooks committed:
FIX remove msg return in find_events, OrderedDict

teonbrooks committed:
MV interest_area interest_areas

dengemann commented:
@teonlamont where are we here actually? Is this still alive?

teonlamont commented:
yep, it is. it needs unit testing and sample data then it will be ready for review.

teonbrooks commented:
closing PR. these scripts have been incorporated into the project-specific analysis scripts
